connection.go

```diff
@@ -37,12 +38,13 @@ type mysqlConn struct {
 	reset            bool // set when the Go SQL package calls ResetSession
 
 	// for context support (Go 1.8+)
-	watching bool
-	watcher  chan<- context.Context
-	closech  chan struct{}
-	finished chan<- struct{}
-	canceled atomicError // set non-nil if conn is canceled
-	closed   atomicBool  // set when conn is closed, before closech is closed
+	watching     bool
+	watcher      chan<- context.Context
+	closech      chan struct{}
+	finished     chan<- struct{}
+	canceled     atomicError // set non-nil if conn is canceled
+	closed       atomicBool  // set when conn is closed, before closech is closed
+	connectionId *uint32
```

Comment:
I don't like pointer. Please use int64 and -1 for undefined.

Reply:
Done